doc/source/whatsnew/v1.0.0.rst

```diff
@@ -176,7 +176,7 @@ Groupby/resample/rolling
 Reshaping
 ^^^^^^^^^
 
--
+- :meth:`DataFrame.merge` now preserves right frame's row order when executing a right merge (:issue:`27453`)
```

Comment:
this needs its own sub-section in api breaking, show the previous and the new

Reply:
Can you address this comment? Want to show the old and new behavior - you can look at previous whatsnew docs for examples of how to show old / new comparisons